File: C/Selecao/1040.c
/**
 * @file 1040.c
 * @author michaelthierry86@gmail.com
 * @brief Média 3: Esse programa lê as 4 notas de um aluno e calcula a media
 * Se a media estiver entre 5 e 7 o aluno tera que faze um exame e sua media é recalculado
 * Se a media for maior do que 7 passa se for menor do que 5 reprova
 * @version 0.1
 * @date 2022-04-18
 * 
 * @copyright Copyright (c) 2022
 * 
 */
/*Inclui Biblioteca*/
#include <stdio.h>
/*Função principal*/
int main(){
    //Declarando Variaves
    double nota_1, nota_2, nota_3, nota_4, media;
    //Lendo do usuario
    scanf("%lf %lf %lf %lf", &nota_1, &nota_2, &nota_3, &nota_4);
    //Calculando media
    media = (nota_1 * 2 + nota_2 * 3 + nota_3 * 4 + nota_4) / 10;
    //Mostra a media
    printf("Media: %.1f\n", media);
    //Avalia se aluno vai ter que fazer o exame
    if(media >= 7.0){
        printf("Aluno aprovado.\n");

    }else if(media >= 5.0){
        //Declarando variavel
        double exame;
        //Mostra que o aluno esta de exame
        printf("Aluno em exame.\n");
        //Le a nota do exame
        scanf("%lf", &exame);
        //Mostra a nota do Exame
        printf("Nota do exame: %.1f\n", exame);
        //Avalia e o aluno passou
        if((exame + media / 2.0) > 5.0){
            //Mostra a mesagem
            printf("Aluno aprovado.\n");

        }else{//senao
            //Mostra a mensagem
            printf("Aluno reprovado.\n");
        }
        //Mostra a media final
        printf("Media final: %.1lf\n", (exame + media) / 2.0);

    }else{//senão
        //Mostra a mensagem
        printf("Aluno reprovado.\n");
    }
    //Fim do programa
    return 0;
}

File: Python/selecao/1040.py
"""
Problema: 1040 - Media 3
"""
#Otendo os dados
nota_1, nota_2, nota_3, nota_4 = input().split(" ")
#Convertendo os valores
nota_1 = float(nota_1)
nota_2 = float(nota_2)
nota_3 = float(nota_3)
nota_4 = float(nota_4)
#calculando
media = float(((nota_1 * 2.0) + (nota_2 * 3.0) + (nota_3 * 4.0) + nota_4) / 10.0)
#Mostrando a media
print("Media: {:.1f}".format(media))
#Decidindo resposta
if media >= 7.0:
    print("Aluno aprovado.")
elif media >= 5.0:
    print("Aluno em exame.")
    exame = float(input())
    print("Nota do exame: {:.1f}".format(exame))
    if ((exame + media) / 2.0) > 5.0:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    nova_media = (exame + media) / 2.0
    print("Media final: {:.1f}".format(nova_media))
else:
    print("Aluno reprovado.")
